File: post-install/app/js/os-release.js
// Citește /etc/os-release la runtime, ca UI-ul să afișeze numele real al
// distribuției instalate (ex: "pearOS Goldwing") în loc de un nume hardcodat.
(function () {
  var fs = require('fs');

  function parseOsRelease(content) {
    var data = {};
    content.split('\n').forEach(function (line) {
      var m = line.match(/^([A-Z_]+)=(.*)$/);
      if (!m) return;
      var value = m[2].trim();
      if (value.length >= 2 && value[0] === '"' && value[value.length - 1] === '"') {
        value = value.slice(1, -1);
      }
      data[m[1]] = value;
    });
    return data;
  }

  function readOsRelease() {
    var paths = ['/etc/os-release', '/usr/lib/os-release'];
    for (var i = 0; i < paths.length; i++) {
      try {
        return parseOsRelease(fs.readFileSync(paths[i], 'utf8'));
      } catch (e) {}
    }
    return {};
  }

  var raw = readOsRelease();

  var OSRelease = {
    id: raw.ID || 'pearos',
    name: raw.NAME || 'pearOS',
    prettyName: raw.PRETTY_NAME || raw.NAME || 'pearOS',
    version: raw.VERSION || '',
    versionCodename: raw.VERSION_CODENAME || '',
    buildId: raw.BUILD_ID || '',
    imageId: raw.IMAGE_ID || '',
    imageVersion: raw.IMAGE_VERSION || '',
    logo: raw.LOGO || '',
    homeUrl: raw.HOME_URL || '',
    documentationUrl: raw.DOCUMENTATION_URL || '',
    supportUrl: raw.SUPPORT_URL || '',
    bugReportUrl: raw.BUG_REPORT_URL || ''
  };

  // Numele hardcodat cu care a fost livrat inițial acest installer, ca să
  // poată fi înlocuit oriunde apare cu numele real al distribuției curente.
  var LEGACY_NAME_PATTERN = /pearOS\s+NiceC0re/g;

  OSRelease.rebrand = function (text) {
    if (typeof text !== 'string') return text;
    return text.replace(LEGACY_NAME_PATTERN, OSRelease.prettyName);
  };

  window.OSRelease = OSRelease;
})();

File: post-install/gtkapp/osrelease.py
"""Port of app/js/os-release.js: read /etc/os-release at runtime so the UI
shows the real installed distro name instead of a hardcoded one."""
import re

_LEGACY_NAME_PATTERN = re.compile(r"pearOS\s+NiceC0re")

_PATHS = ["/etc/os-release", "/usr/lib/os-release"]


def _parse_os_release(content):
    data = {}
    for line in content.split("\n"):
        m = re.match(r"^([A-Z_]+)=(.*)$", line)
        if not m:
            continue
        value = m.group(2).strip()
        if len(value) >= 2 and value[0] == '"' and value[-1] == '"':
            value = value[1:-1]
        data[m.group(1)] = value
    return data


def _read_os_release():
    for path in _PATHS:
        try:
            with open(path, "r", encoding="utf-8") as f:
                return _parse_os_release(f.read())
        except OSError:
            continue
    return {}


class _OSRelease:
    def __init__(self):
        raw = _read_os_release()
        self.id = raw.get("ID", "pearos")
        self.name = raw.get("NAME", "pearOS")
        self.pretty_name = raw.get("PRETTY_NAME", raw.get("NAME", "pearOS"))
        self.version = raw.get("VERSION", "")
        self.version_codename = raw.get("VERSION_CODENAME", "")
        self.build_id = raw.get("BUILD_ID", "")
        self.image_id = raw.get("IMAGE_ID", "")
        self.image_version = raw.get("IMAGE_VERSION", "")
        self.logo = raw.get("LOGO", "")
        self.home_url = raw.get("HOME_URL", "")
        self.documentation_url = raw.get("DOCUMENTATION_URL", "")
        self.support_url = raw.get("SUPPORT_URL", "")
        self.bug_report_url = raw.get("BUG_REPORT_URL", "")

    def rebrand(self, text):
        if not isinstance(text, str):
            return text
        return _LEGACY_NAME_PATTERN.sub(self.pretty_name, text)


OS_RELEASE = _OSRelease()
